Sources/Architecture/X86/HardwareBreakpointManager.cpp

```diff
@@ -184,6 +184,29 @@ int HardwareBreakpointManager::getAvailableLocation() {
 
   return (it - _locations.begin());
 }
+
+ErrorCode HardwareBreakpointManager::isValid(Address const &address,
+                                             size_t size, Mode mode) const {
+  switch (size) {
+  case 1:
+    break;
+  case 8:
+    DS2LOG(Warning, "8-byte breakpoints not supported on all architectures");
```

Comment:
We talked about this before haha, it's more complicated than just x86 vs x86_64:

> For Pentium® 4 and Intel® Xeon® processors with a CPUID signature corresponding to family 15 (model 3, 4, and 6), break point conditions permit specifying 8-byte length on data read/write with an of encoding 10B in the LENn field.
Encoding 10B is also supported in processors based on Intel Core microarchitecture or enhanced Intel Core microarchitecture, the respective CPUID signatures corresponding to family 6, model 15, and family 6, DisplayModel value 23 (see CPUID instruction in Chapter 3, “Instruction Set Reference, A-M” in the Intel® 64 and IA-32 Architectures Software Developer’s Manual, Volume 2A). The Encoding 10B is supported in processors based on Intel® AtomTM microarchitecture, with CPUID signature of family 6, DisplayModel value 1CH. The encoding 10B is undefined for other processors.

>